lib/services/search.js

```diff
@@ -0,0 +1,404 @@
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @param {object} source
+ * @returns {Promise}
+ */
+function put(index, type, ref, source) {
+  return client.index({
+    index: index,
+    type: type,
+    id: ref,
+    body: source
+  }).then(function (resp) {
+    log('info', resp);
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Delete an Elasticsearch document
+ *
+ * @param {string} index
+ * @param {string} type
+ * @param {string} ref, e.g. 'localhost.dev.nymag.biz/scienceofus/components/article/instances/section-test'
+ * @returns {Promise}
+ */
+function del(index, type, ref) {
+  return client.delete({
+    index: index,
+    type: type,
+    id: ref
+  }).then(function (resp) {
+    log('info', resp);
+    return resp;
+  })
+    .catch(function (error) {
+      log('error', error);
+      return bluebird.reject(error);
+    });
+}
+
+/**
+ * Perform multiple index operations
+ *
+ * @param {Array} ops
+ * @returns {Promise}
+ */
+function batch(ops) {
+  return client.bulk({
+    body: ops
+  }).then(function (resp) {
+    if (resp && resp.errors === true) {
+      let str = ['Client.bulk errored on ' + util.inspect(ops, {depth: 10})];
+
+      return bluebird.reject(new Error(str));
+    }
+  });
+}
+
+/**
+ * Create the ES Client or an empty object
+ *
+ * @param {Object} overrideClient
+ */
+function setup(overrideClient) {
+  if (!module.exports.endpoint && !overrideClient) {
+    throw new Error('No Elastic endpoint or client override');
+  }
+
+  // Set the exported client
+  module.exports.client = client = module.exports.endpoint && !overrideClient ? new elastic.Client(serverConfig) : _.assign(module.exports.client, overrideClient);
+}
+
+/**
+ * Sets the alias suffix for use
+ * @param {[type]} suffix [description]
+ */
+function setAliasSuffix(suffix) {
+  if (_.isString(suffix)) {
+    aliasSuffix = suffix;
+  } else {
+    throw new Error('Alias suffix required to be a string');
+  }
+}
+
+module.exports.setup = setup;
+module.exports.setAliasSuffix = setAliasSuffix;
+module.exports.endpoint = endpoint;
+module.exports.healthCheck = healthCheck;
+module.exports.initIndex = initIndex;
+module.exports.initAlias = initAlias;
+module.exports.delIndex = delIndex;
+module.exports.createIndexIfNone = createIndexIfNone;
+module.exports.createAliasIfNone = createAliasIfNone;
+module.exports.existsAlias = existsAlias;
+module.exports.existsIndex = existsIndex;
+module.exports.initMapping = initMapping;
+module.exports.existsMapping = existsMapping;
+module.exports.createMappingIfNone = createMappingIfNone;
+module.exports.createIndexName = createIndexName;
+module.exports.convertRedisBatchtoElasticBatch = convertRedisBatchtoElasticBatch;
+module.exports.del = del;
+module.exports.batch = batch;
+module.exports.put = put;
+module.exports.query = query;
```

Comment:
@larrychevres per our conversation yesterday about aliases and our workflow I have made the following updates:

- Changed the [function](https://github.com/nymag/amphora/pull/327/files#diff-0f9d28e9bf5f9c727a7d43583bea7b3fR361)/[variable](https://github.com/nymag/amphora/pull/327/files#diff-0f9d28e9bf5f9c727a7d43583bea7b3fR18) that were named for setting an alias version to now set an index version.
- Added a check [here](https://github.com/nymag/amphora/pull/327/files#diff-0f9d28e9bf5f9c727a7d43583bea7b3fR362) so that a version must to begin with `_v` for consistency 

Per your comment on checking/updating indices, that logic is over in [sites](https://github.com/nymag/sites/blob/search-service/services/search/index.js#L323-L336) and is what we currently do now.